extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/mpmetrics/AnnotatedGaugeAdapter.java

```diff
@@ -52,7 +53,8 @@ public GaugeAdapterImpl(String name, String description, String targetName, Stri
         public GaugeAdapterImpl register(MetricDescriptor id, MeterRegistry registry) {
             this.id = id;
             if (gauge == null || metadata.cleanDirtyMetadata()) {
-                gauge = io.micrometer.core.instrument.Gauge.builder(metadata.name, this::getValue)
+                ToDoubleFunction<GaugeAdapterImpl> function = i -> i.getValue().doubleValue();
+                gauge = io.micrometer.core.instrument.Gauge.builder(metadata.name, this, function)
```

Comment:
I still am not sure how the two registries are interacting, but I think it is clear that when both are present, the way the json registry observes the value of the gauge causes the prometheus registry value to observe NaN, and that breaks the planet. ;)  Switching over to invoking a method on the object is fine, but we should avoid allocation of ToDoubleFunction for each instance. We could make that lambda a static, or we could use a method on the object:

        double getDoubleValue() {
            return getValue().doubleValue();
        }

and reference that: 

                    gauge = io.micrometer.core.instrument.Gauge.builder(metadata.name, this, GaugeAdapterImpl::getDoubleValue)